### PR TITLE
Ap 4630 partner/cash income

### PR DESCRIPTION
**Pull request by skdejong**
## What

[AP-4630: partner/cash_income and outgoings - remembers positive answers provided for client](https://dsdmoj.atlassian.net/browse/AP-4630)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- The standards in the [Git Workflow document on Confluence](https://dsdmoj.atlassian.net/wiki/spaces/ATPPB/pages/4602855954/Git+Workflow) should be followed
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/views/shared/means/_cash_income.html.erb [hunk 1]
```diff
@@ -26,7 +26,7 @@
       </div>
 
       <%= form.govuk_radio_divider %>
-      <%= form.govuk_check_box :none_selected, true, "", multiple: false, label: { text: none_of_the_above }, checked: @legal_aid_application.no_cash_income? %>
+      <%= form.govuk_check_box :none_selected, true, "", multiple: false, label: { text: none_of_the_above }, checked: type.present? ? type.no_cash_income? : @legal_aid_application.no_cash_income? %>
```
Comment (colinbruce) [inline app/views/shared/means/_cash_income.html.erb:29]: As you have updated the means versions to set type as @legal_aid_application you can remove the ternary check and just use 
```suggestion
      <%= form.govuk_check_box :none_selected, true, "", multiple: false, label: { text: none_of_the_above }, checked: type.no_cash_income? %>
```

File: app/views/shared/means/_cash_outgoing.html.erb [hunk 2]
```diff
@@ -27,7 +27,7 @@
       </div>
 
       <%= form.govuk_radio_divider %>
-      <%= form.govuk_check_box :none_selected, true, "", multiple: false, label: { text: t("generic.none_of_the_above") }, checked: @legal_aid_application.no_cash_outgoings? %>
+      <%= form.govuk_check_box :none_selected, true, "", multiple: false, label: { text: t("generic.none_of_the_above") }, checked: type.present? ? type.no_cash_outgoings? : @legal_aid_application.no_cash_outgoings? %>
```
Comment (colinbruce) [inline app/views/shared/means/_cash_outgoing.html.erb:30]: ditto not needing the type.present check

File: db/schema.rb [hunk 3]
```diff
@@ -1074,8 +1076,8 @@
   add_foreign_key "legal_aid_applications", "providers"
   add_foreign_key "legal_framework_merits_task_lists", "legal_aid_applications", on_delete: :cascade
   add_foreign_key "legal_framework_submissions", "legal_aid_applications"
-  add_foreign_key "linked_applications", "legal_aid_applications", column: "associated_application_id"
-  add_foreign_key "linked_applications", "legal_aid_applications", column: "lead_application_id"
+  add_foreign_key "linked_applications", "legal_aid_applications", column: "associated_application_id", validate: false
```
Comment (colinbruce) [inline db/schema.rb:1079]: this has crept in from a rails db:migrate and has nothing to do with you code change... let's pair to get these removed :+1:

**Review by colinbruce — COMMENTED** [review]
Looking good - a few minor tweaks but overall... ⭐

**Review by RoseSAK — APPROVED** [review]
Looks good! 🎉

**Pull request merged**